### PR TITLE
Fix postgresql lib support and upgrade conflicts.

### PR DESCRIPTION
**Pull request by gothrek22**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,18 +1,39 @@
...
+[pgdg-common]
+name=PostgreSQL common RPMs for RHEL / Rocky / AlmaLinux \$releasever - \$basearch
+baseurl=https://download.postgresql.org/pub/repos/yum/common/redhat/rhel-\$releasever-\$basearch
+enabled=1
+gpgcheck=1
+
+[pgdg-rhel9-sysupdates]
+name=PostgreSQL Supplementary ucommon RPMs for RHEL / Rocky / AlmaLinux \$releasever - \$basearch
+baseurl=https://download.postgresql.org/pub/repos/yum/common/pgdg-rocky9-sysupdates/redhat/rhel-\$releasever-\$basearch
+enabled=0
```
Comment (mb-shorai) [inline Dockerfile]: I guess this repo can be removed if it's disabled?

File: Dockerfile [hunk 2]
```diff
@@ -1,18 +1,39 @@
...
+baseurl=https://download.postgresql.org/pub/repos/yum/${POSTGRES_VERSION}/redhat/rhel-\$releasever-\$basearch
+enabled=1
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/PGDG-RPM-GPG-KEY-RHEL
+EOF
+
+RUN curl -o /etc/pki/rpm-gpg/PGDG-RPM-GPG-KEY-RHEL https://download.postgresql.org/pub/repos/yum/keys/PGDG-RPM-GPG-KEY-RHEL && \
     microdnf --nodocs -y install epel-release && \
     microdnf module enable -y "ruby:${RUBY_VERSION}" && \
-    rpm -i "https://download.postgresql.org/pub/repos/yum/reporpms/EL-${ROCKY_VERSION}-$(arch)/pgdg-redhat-repo-latest.noarch.rpm" && \
```
Comment (mb-shorai) [inline Dockerfile:13]: PSQL actually recommends this method in their docs, why switch to hardcoded repo definitions and download the key manually?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -36,4 +36,4 @@ jobs:
 
       - name: Push images
         if: github.ref == 'refs/heads/master' && github.event_name != 'pull_request'
-        run: docker buildx bake --push
+        run: docker buildx bake --push --no-cache
```
Comment (mb-shorai) [inline .github/workflows/main.yml]: Why no cache? The `Build images` step above doesn't use cache anyway, so `docker buildx bake --push` is efficient and always uses fresh cache from the previous step.

File: Dockerfile (hunk 2 above)
Comment (gothrek22) [reply Dockerfile:13]: The issue with that approach is that by default they enable ALL versions of postgresql that they supply. In testing I've noticed that this causes issues with upgrading packages and unneeded hassle. The end result is more or less the same, as the structure of their repo hasn't changed in years.

File: .github/workflows/main.yml (hunk 3 above)
Comment (gothrek22) [reply .github/workflows/main.yml]: Leftover from testing, good catch.

**Review by mb-shorai — APPROVED** [review]

**Review by mparadowsky — APPROVED** [review]

**Pull request merged**